### DISCUSSION TITLE
cli `print-schema` support sqlite `rowid` without explicit `PRIMARY KEY`

### DISCUSSION THREAD

**Discussion by hasezoey** [diesel-rs/diesel#3679] [Ideas]
sqlite tables always have a `rowid`, `oid` and `_rowid_`, even when not specifying `PRIMARY KEY` on any column, as long as `WITHOUT ROWID` is not used. but currently `print-schema` errors with `Diesel only supports tables with primary keys` if a table without explicit `PRIMARY KEY` is encountered

[sqlite CREATE TABLE ROWIDs and the INTEGER PRIMARY KEY](https://www.sqlite.org/lang_createtable.html#rowids_and_the_integer_primary_key)

example SQL:

```sql
CREATE TABLE test(
	test INTEGER
);
	
INSERT INTO test(test) VALUES (NULL), (1);

SELECT * FROM test WHERE rowid = 2;
```

diesel 
 Version: 2.1.0
 Supported Backends: sqlite

installed via `cargo install diesel_cli --no-default-features --features sqlite`

my implementation "expectation" would be that `print-schema` does not error for such cases and automatically adds a column named `rowid` or any of the aliases as the primary key unless a explicit `PRIMARY KEY` is used or error if no `PRIMARY KEY` is present and `WITHOUT ROWID` is used

**Comment by weiznich** [comment]
I'm happy to accept a PR for that as long as:

* it contains tests + documentation
* it does not break existing code
